.githooks/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+# This is a simple check to ensure people remember to regenerate the docs
+# and the pb.go files if they change a .proto file.
+# It's not iron-clad but it's better than finding out later in github.
+
+# AMD = added, modified, deleted
+files=$(git diff --cached --name-only --diff-filter=AMD)
+
+LIGHT_RED='\033[1;31m'
+RESET='\033[0m' # No Color
+
+# naive regex check for .proto extension
+if [[ "$files" =~ \.proto ]]
```

Comment:
Why are we checking if the proto files where changed?

You should run `make` in the CI and see if the result of that is _any_ changed file anywhere through `git diff-files --quiet`
if the exit code is 0, that means no changes and the docs match what should be generated, if the exit code is 1 then there are uncommited changes, so error